tests/gold_tests/pluginTest/certifier/certifier.test.py

```diff
@@ -0,0 +1,169 @@
+
+    def setupTS(self):
+        self.ts = Test.MakeATSProcess("ts1", enable_tls=True)
+        self.ts.addDefaultSSLFiles()
+        # copy over the cert store in which the certs will be generated/stored
+        self.certPathDest = os.path.join(self.ts.Variables.CONFIGDIR, "certifier-certs")
+        Setup.Copy(self.certPathSrc, self.certPathDest)
+        self.ts.Disk.records_config.update({
+            "proxy.config.diags.debug.enabled": 1,
+            "proxy.config.diags.debug.tags": "http|certifier|ssl",
+            "proxy.config.ssl.server.cert.path": f'{self.ts.Variables.SSLDir}',
+            "proxy.config.ssl.server.private_key.path": f'{self.ts.Variables.SSLDir}',
+        })
+        self.ts.Disk.ssl_multicert_config.AddLine(
+            'dest_ip=* ssl_cert_name=server.pem ssl_key_name=server.key'
+        )
+        self.ts.Disk.remap_config.AddLine(
+            f"map / http://127.0.0.1:{self.server.Variables.http_port}/",
+        )
+        self.ts.Disk.plugin_config.AddLine(
+            f'certifier.so -s {os.path.join(self.certPathDest, "store")} -m 1000 -c {os.path.join(self.certPathDest, "ca.cert")} -k {os.path.join(self.certPathDest, "ca.key")} -r {os.path.join(self.certPathDest, "ca-serial.txt")}')
+        # Verify logs for dynamic generation of certs
+        self.ts.Disk.traffic_out.Content += Testers.ContainsExpression(
+            "Creating shadow certs",
+            "Verify the certifier plugin generates the certificate dynamically.")
+
+    def runHTTPSTraffic(self):
+        tr = Test.AddTestRun("Test dynamic generation of certs")
+        tr.AddVerifierClientProcess(
+            "client1",
+            self.httpsReplayFile,
+            http_ports=[self.ts.Variables.port],
+            https_ports=[self.ts.Variables.ssl_port],
+            other_args='--thread-limit 1')
+        tr.Processes.Default.StartBefore(self.server)
+        tr.Processes.Default.StartBefore(self.ts)
+        tr.StillRunningAfter = self.server
+        tr.StillRunningAfter = self.ts
+
+    def verifyCert(self, certPath):
+        tr = Test.AddTestRun("Verify the content of the generated cert")
+        tr.Processes.Default.Command = f'openssl x509 -in {certPath} -text -noout'
+        tr.Processes.Default.ReturnCode = 0
+        # Verify certificate content
+        tr.Processes.Default.Streams.All += Testers.ContainsExpression(
+            "Subject: CN = www.tls.com", "Subject should match the host in the request")
+        tr.Processes.Default.Streams.All += Testers.ContainsExpression(
+            r"X509v3 extensions:\n.*X509v3 Subject Alternative Name:.*\n.*DNS:www.tls.com",
+            "Should contain the SAN extension",
+            reflags=re.MULTILINE)
+
+    def verifyCertNotExist(self, certPath):
+        tr = Test.AddTestRun("Verify the cert doesn't exist in the store")
+        tr.Processes.Default.Command = "echo verify"
+        certFile = tr.Disk.File(certPath)
+        certFile.Exists = False
```

Comment:
I think this is undocumented, but `File` takes an `exists` parameter in the constructor. Let's use that instead of two lines here:

https://bitbucket.org/autestsuite/reusable-gold-testing-system/src/01eacd6796f42e063c35e5c0005bf5912d767e6e/src/autest/testenities/file.py#lines-24

```
certFile = tr.Disk.File(certPath, exists=False)
```

(This might sound like a super trivial nit., but since tests get copy-and-pasted a lot, it's good to set this as an example for other tests.)

Reply:
Thanks for pointing this out. Modified that and looks cleaner.